src/transformers/generation_utils.py

```diff
@@ -1326,6 +1333,41 @@ def generate(
                 **model_kwargs,
             )
 
+        elif is_constraint_gen_mode:
```

Comment:
If I understand correctly, this makes only sense if:
- `num_beams` > 1 
- `do_sample = False`

no? 

Could we maybe raise a `ValueError` if any of the above conditions is not fullfilled? This would also greatly help us reduce maintenance in a first step since we're limiting the amount of possible use cases. We could always later add contrained generation for sampling or non-beam search. What do you think?

Reply:
Agreed. I'll adjust accordingly in a following commit.